### DISCUSSION TITLE
Form error still displayed when fixed for react-hook-form-register-with-error-messages example

### DISCUSSION THREAD

**Discussion by loicbourg** [react-hook-form/react-hook-form#4931] [Q&A]
**Describe the bug**
on https://codesandbox.io/s/react-hook-form-getvalues-compare-field-values-orf0p?file=/src/index.js

when i resolve the "Passwords should match!" error by editing the new password field, the error is not removed on change

**To Reproduce**

1. Go to https://codesandbox.io/s/react-hook-form-getvalues-compare-field-values-orf0p?file=/src/index.js
2. write "A" in "New Password" input
3. write "AA" in "New Password" input
4. Click on the "trigger" button
5. write "AA" in "New Password" input

**Codesandbox link**
https://codesandbox.io/s/react-hook-form-getvalues-compare-field-values-orf0p?file=/src/index.js

**Expected behavior**
The error message "Passwords should match!" should disapear when i write the value "AA" in new password

**Screenshots**
![Capture d’écran de 2021-04-22 12-46-47](https://user-images.githubusercontent.com/4905942/115702511-52254580-a369-11eb-83ef-1002fe3f6eab.png)

**Desktop (please complete the following information):**

- OS: Debian 10
- Browser Chrome
- Version 89

**Additional context**

I tried to change the mode of useForm to "onBlur" but I have the same error since the validation is only triggered when the confirm password field has an event.

**Comment by bluebill1049** [comment]
<img width="1387" alt="Screen Shot 2021-04-22 at 9 09 43 pm" src="https://user-images.githubusercontent.com/10513364/115704732-11501f80-a3af-11eb-8d8e-48e4d599d48d.png">

can you test with the latest version?

**Comment by loicbourg** [comment]
This is the latest react hook form (7.2.3)

I made a video to show you the steps:
![output](https://user-images.githubusercontent.com/4905942/115719095-0f209d80-a37c-11eb-9d55-7fd582a747f3.gif)

**Comment by bluebill1049** [comment]
<img width="343" alt="Screen Shot 2021-04-23 at 11 46 01 am" src="https://user-images.githubusercontent.com/10513364/115806096-7c3e3c80-a429-11eb-8d4e-e28ebf783acb.png">

but you are on mode: 'onSubmit'? you haven't submitted the form right? if you want to have validation onChange, use `onChange` mode.

**Comment by loicbourg** [comment]
When the validation is onSubmit mode, changing the value of the "Confirm password" field fix the error when it is displayed.

When i edit the "new password" field, i would expect the error to be fixed too.

i tried to use the reValidateMode: "onChange" but I have the same behavior.

![Capture d’écran de 2021-04-23 15-45-34](https://user-images.githubusercontent.com/4905942/115880459-0ba81780-a44b-11eb-8b8a-bfa698bcc192.png)

**Reply by bluebill1049** [reply]
those are the default mode:
<img width="975" alt="Screen Shot 2021-04-23 at 11 49 51 pm" src="https://user-images.githubusercontent.com/10513364/115880956-99f1bd00-a48e-11eb-91b4-74b47f721f8f.png">

I think you want mode to set to 'onChange'.

**Reply by bluebill1049** [reply]
mode: 'onChange' will trigger validation on each key stroke, that's what you want right?

**Reply by loicbourg** [reply]
@bluebill1049 not exactly, my problem is that I don't see how to fix the confirm password error when editing the new password field.

![Capture d’écran de 2021-04-23 15-52-58](https://user-images.githubusercontent.com/4905942/115881400-ff708a00-a44b-11eb-89aa-77e105e8ddfb.png)

**Reply by bluebill1049** [reply]
Do you mean the first input? you may need to trigger the validation? because the first input doesn't have validation associated with it.

**Reply by loicbourg** [reply]
Yes that's what i am trying to do, i think it's weird for the user to have the error fixed onChange for the second input and onSubmit for the first.

The solution would be to have a useEffect on the newPassword value and execute triggerValidation  ?

**Reply by bluebill1049** [reply]
It's late here, let me give it a try tmr morning. I don't think it's bug with the lib, it's more of how to achieve it nicely.

**Reply by loicbourg** [reply]
No problem, thank you for your answers !
